diff --git a/wgengine/userspace.go b/wgengine/userspace.go
index 4a9f32143..a660f9e00 100644
--- a/wgengine/userspace.go
+++ b/wgengine/userspace.go
@@ -538,7 +538,7 @@ func NewUserspaceEngine(logf logger.Logf, conf Config) (_ Engine, reterr error)
 		conf.SetSubsystem(e.tundev)
 		conf.SetSubsystem(e.magicConn)
 		conf.SetSubsystem(e.dns)
-		conf.SetSubsystem(conf.Router)
+		// conf.SetSubsystem(conf.Router)
 		conf.SetSubsystem(conf.Dialer)
 		conf.SetSubsystem(e.netMon)
 		if conf.DriveForLocal != nil {
